pancancer_evaluation/data_models/tcga_data_model.py

```diff
@@ -509,3 +637,77 @@ def holdout_percent_labels(y,
             test_ixs |= nz_ixs
         return (train_ixs, test_ixs)
 
+    @staticmethod
+    def _get_cancers_to_add(y_df,
+                            test_cancer_type,
+                            num_cancer_types,
+                            how_to_add,
+                            similarity_matrix=None,
+                            seed=cfg.default_seed):
+
+        # start with test cancer type and add if necessary
+        train_cancer_types = [test_cancer_type]
+        # y_df should already be filtered to valid cancer types
+        valid_cancer_types = [ct for ct in y_df.DISEASE.unique()
+                                 if ct != test_cancer_type]
+
+        if num_cancer_types == -1:
+            # pan-cancer model, train on all valid cancer types
+            train_cancer_types += valid_cancer_types
+        elif num_cancer_types >= 1:
+            # add desired number of cancer types to train set
+            if how_to_add == 'random':
+                import contextlib
+                # We want this random addition of cancer types to be the same
```

Comment:
I'm not sure if i understand the logic for why you're re-seeding here. So each time you're adding a random cancer type so that you have target + A or target + A,B or target + A, B, C. And each time you want to preserve the previously added cancers + adding a new random one. Why doesn't using a global see and say passing in a seed to select the additional one cancer work?

Reply:
I think this is a duplicate of your first comment (let me know if I'm wrong).

Reply:
yup..i was moving around tabs and refreshing things so you can just ignore this. I thought the comment was lost when i closed the tab